oauthenticator/oauth2.py

```diff
@@ -316,10 +315,62 @@ def _validate_server_cert_default(self):
         else:
             return True
 
+    http_client = Any()
+
+    @default("http_client")
+    def _default_http_client(self):
+        return AsyncHTTPClient()
```

Comment:
This may be called a bit more frequently and not reused within the scope of handling requests, but I presume it will be fine. Do you agree?

Reply:
This will be called significantly *less* frequently, as it once per Authenticator instantiation (i.e. once per process) rather than once per method call.

However, due to how tornado 'instance' metaclasses work, each call to `AsyncHTTPClient()` actually returns the global instance by default, so there should still be just one Client object.